crawler/dockercontainer.py

```diff
@@ -72,13 +87,27 @@ def poll_docker_containers(timeout, user_list=None, host_namespace=''):
     containers.
     :return: a DockerContainer object (just the first container created).
     """
-
-    """
-    TODO: implement this function. For now, just emulate a failed poll:
-    no containers were created before timeout seconds.
-    """
     if timeout > 0:
-        time.sleep(timeout)
+        eventQ = multiprocessing.JoinableQueue()
+        eventProc = multiprocessing.Process(
+            name='docker-monitor',
+            target=start_polling, args=(eventQ, ))
```

Comment:
What about something like this instead of starting a new process:

```
def poll_containers(timeout, user_list=None, host_namespace=''):
    with timeout(seconds=timeout):
         events = self.client.events(decode=True)
         for event in events:
                 if event['Action'] == 'start':
                        container = DockerContainer(long_id=event['id'], host_namespace=host_namespace)
                        self.crawl_container(container)
```

You can reuse that `timeout` wrapper from here: http://stackoverflow.com/questions/2281850/timeout-function-if-it-takes-too-long-to-finish